tests/foreman/api/test_classparameters.py

```diff
@@ -0,0 +1,1035 @@
+        7.  Set 'merge default' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output doesn't have the puppet default value.
+        3.  Duplicate values in YAML output if any are displayed.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_create_matcher_avoid_duplicate(self):
+        """Merge the values of all the associated matchers, remove duplicates.
+
+        @feature: Class Parameters - Matcher
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set some default Value of array type.
+        3.  Create first matcher for attribute fqdn with some value.
+        4.  Create second matcher for other attribute with
+        same value as fqdn matcher.
+        Note - The fqdn/host should have this attribute.
+        5.  Set 'merge overrides' to True.
+        6.  Set 'merge default' to True.
+        7.  Set 'avoid duplicate' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all
+        the associated matchers.
+        2.  The YAML output has the default value of parameter.
+        3.  Duplicate values in YAML output are removed / not displayed.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_create_matcher_avoid_duplicate(self):
+        """Duplicates not removed as they were not really present.
+
+        @feature: Class Parameters - Matcher
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set some default Value of array type.
+        3.  Create first matcher for attribute fqdn with some value.
+        4.  Create second matcher for other attribute with other value
+        than fqdn matcher and default value.
+        Note - The fqdn/host should have this attribute.
+        5.  Set 'merge overrides' to True.
+        6.  Set 'merge default' to True.
+        7.  Set 'merge avoid duplicates' to True.
+        8.  Update the parameter with above steps.
+        9.  Go to YAML output of associated host.
+
+        @assert:
+
+        1.  The YAML output has the values merged from all matchers.
+        2.  The YAML output has the default value of parameter.
+        3.  No value removed as duplicate value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_enable_merge_overrides_default_checkboxes(self):
+        """Enable Merge Overrides, Merge Default checkbox for supported types.
+
+        @feature: Class Parameters - Matcher
+
+        @steps:
+
+        1.  Set parameter type to array/hash.
+
+        @assert: The Merge Overrides, Merge Default checks are
```

Comment:
I thinks its already formatted well. Can you please give an example of proper formatting.